### ISSUE TITLE
docker image support for pod only includes docker hub

### ISSUE THREAD

**Issue by deitch** [lf-edge/eden#210]
If I try to deploy `eden pod deploy docker://quay.io/coreos/etcd:v3.4.10`, or, for that matter, `eden pod deploy docker://docker.io/library/nginx:latest`, it fails.

The core issue is that eden assumes that `docker://` _always_ means docker hub, and everything after `docker://` is the path. This also violates URI naming conventions.

This should be changed so that:

* everything after `://` is a normal URI, i.e. the first part is the host
* if no host is provided, we can default to `docker.io`, but I don't think we should; I think it should error out

We might want to consider changing `docker:` to `oci:`, since it is oci registries, but that is a nitpick, and we can stay with the colloquial `docker:` if we want.

**Comment by rvs** [comment]
This is a great point @deitch !

In fact, let me through on suggestion on top of what you're saying. How about:

```
oci://docker.io/nginx[:latest]
oci:///nginx[:latest] == oci://docker.io/nginx[:latest]
```

and we keep docker the way it is today? IOW to be:

```
docker://nginx[:latest] == oci://docker.io/nginx[:latest]
```

Or is it too confusing and we should just stick with the oci:// ?

**Comment by rvs** [comment]
I just realized that this is what you did in your latest PR ;-) great minds I suppose and all that ;-)

Should this issue be closed then @deitch ?

**Comment by deitch** [comment]
Funny, I thought it had closed it. Definitely 

Closed via #215
